src/chainstate/burn/mod.rs

```diff
@@ -181,6 +185,18 @@ impl SortitionHash {
         SortitionHash(ret)
     }
 
+    /// Choose n indices (without replacement) from the range [0, max).
+    pub fn choose(&self, n: u32, max: u32) -> Vec<u32> {
```

Comment:
Won't this algorithm occasionally return duplicate entries in the resulting `Vec<u32>`?  I'm not seeing how the "without replacement" aspect is being guaranteed here.

Reply:
This function will always return `n` unique indices: that's the guarantee of the [`sample`](https://docs.rs/rand/0.7.3/rand/seq/index/fn.sample.html).

That recipient info object is consumed by the `add_fork_info...` function to swap out any consumed indices from the stored reward set (and reduce the reward set size).